### PR TITLE
code cleanup and user controller changes and update dto renaming (1.0)

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
1). did code cleanup with removing redundant logic
2). user controller was updated 
3). update.course.dto renamed to avoid confusion
4). wishlist tested with empty array
5). user creation tested with Postman
6). API property added
7). At the moment, those not required references from schemas removed
8). certain service naming requiring camel casing fixed
9). put the schemas inside the respected models in a schema folder to avoid confuion

**Type of Change:**
- [x] Code

**How Has This Been Tested?**
1).npm start
2).Response check using Postman

### CODE REVIEW

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 Please suggest changes for this PR or changes that can be added in the followup for this

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

File: src/firebase/firebase.module.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { Module } from '@nestjs/common';
-import { FirebaseAdminCoreModule } from '@tfarras/nestjs-firebase-admin';
+import { FirebaseAdminModule } from '@tfarras/nestjs-firebase-admin';
```
[the remark this answers is not in the document]
Comment (Devesh21700Kumar) [reply src/firebase/firebase.module.ts:2]: Yes ,it got deprecated

File: src/models/assignment/assignment.controller.ts [hunk 2]
```diff
@@ -34,50 +33,37 @@ export class AssignmentController {
...
+    return await this.AssignmentService.getAllAssignment();
   }
 
   // Fetch a particular Assignment using ID
   @ApiCreatedResponse({ type: AssignmentResponseBody })
   @Get('/:assignmentId')
   async getAssignment(@Param('assignmentId') AssignmentId: string) {
-    const assignment = await this.AssignmentService.getAssignment(AssignmentId);
-    return assignment;
+    return await this.AssignmentService.getAssignment(AssignmentId);
```
[the remark this answers is not in the document]
Comment (Devesh21700Kumar) [reply src/models/assignment/assignment.controller.ts:50]: Ok

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Put all schemaas into respected models.
create a folder schema inside each model and put the schema file there.

**Comment by Devesh21700Kumar** [comment]
> Put all schemaas into respected models.
> create a folder schema inside each model and put the schema file there.

I've made the schema folder in each model and added the schema file inside it and tested them again

**Pull request merged**